Diagnostics/FindFrontEndActivity.ps1

```diff
@@ -0,0 +1,198 @@
+    entries. This may be somewhat faster. However, there is no filtering for LatencyThreshold
+    and NotificationWait when this option is used.
+.PARAMETER TimeSpan
+    Specify how far back to search the logs. This is a TimeSpan value, such as "01:00" for the
+    last hour or "00:30" for the last 30 minutes. (Default: 15 minutes). Use this parameter to
+    search the most recent logs. Use StartTime and EndTime to search older logs.
+.PARAMETER StartTime
+    Logs older than this time are not searched. This is a DateTime value, such as (Get-Date).AddDays(-1)
+    or "2023-02-11 08:00". Use this parameter to search old logs. Use -TimeSpan to search the
+    most recent logs.
+.PARAMETER EndTime
+    Logs newer than this time are not searched. This is a DateTime value, such as (Get-Date).AddDays(-1)
+    or "2023-02-11 09:00". Use this parameter to search old logs. Use -TimeSpan to search the
+    most recent logs.
+.LINK
+    https://microsoft.github.io/CSS-Exchange/Diagnostics/FindFrontEndActivity
+.EXAMPLE
+    Get-ExchangeServer | .\FindFrontEndActivity.ps1 -SamAccountName "user1", "user2" | ft
+    Show any MAPI HttpProxy activity that took more than 1 second for user1 or user2 within the last 15 minutes on all Exchange servers in the forest.
+.EXAMPLE
+    Get-ExchangeServer | .\FindFrontEndActivity.ps1 -SamAccountName "user1", "user2" -Quiet
+    Show only the server names where user1 or user2 are found in the MAPI HttpProxy logs within the last 15 minutes.
+.EXAMPLE
+    Get-ExchangeServer | .\FindFrontEndActivity.ps1 -SamAccountName "user1", "user2" -Protocol "ews", "mapi" -LatencyThreshold 100 -TimeSpan "00:30"
+    Show any EWS or MAPI HttpProxy activity that took more than 100 milliseconds for user1 or user2 within the last 30 minutes on all Exchange servers in the forest.
+#>
+[CmdletBinding(DefaultParameterSetName = 'Recent')]
+param (
+    [Parameter(Mandatory = $true, ValueFromPipeline = $true, ValueFromPipelineByPropertyName = $true)]
+    [Alias('Fqdn')]
+    [string[]]
+    $ServerName,
+
+    [Parameter(Mandatory = $true)]
+    [string[]]
+    $SamAccountName,
+
+    [Parameter()]
+    [int]
+    $LatencyThreshold = 1000,
+
+    [Parameter()]
+    [ValidateSet('Autodiscover', 'EAS', 'ECP', 'EWS', 'MAPI', 'OWA', 'PowerShell', 'RpcHttp')]
+    [string[]]
+    $Protocol = @('MAPI'),
+
+    [Parameter()]
+    [switch]
+    $IncludeNonExecutes,
+
+    [Parameter()]
+    [switch]
+    $Quiet,
+
+    [Parameter(ParameterSetName = 'Recent')]
+    [TimeSpan]
+    $TimeSpan = (New-TimeSpan -Minutes 15),
+
+    [Parameter(ParameterSetName = 'Range', Mandatory = $true)]
+    [DateTime]
+    $StartTime,
+
+    [Parameter(ParameterSetName = 'Range', Mandatory = $true)]
+    [DateTime]
+    $EndTime
+)
+
+begin {
+    $serverNames = New-Object System.Collections.ArrayList
+    $samAccountRegexForFileMatch = [string]::Join('|', ($SamAccountName | ForEach-Object { "\\$_," }))
```

Comment:
Might want to verify that the passed value isn't something like this `contoso\user`